diff -Naur zlib-1.2.8/Makefile zlib-1.2.8-psp/Makefile
--- zlib-1.2.8/Makefile	2011-09-10 02:33:23.000000000 -0400
+++ zlib-1.2.8-psp/Makefile	2014-06-11 21:15:42.530582880 -0400
@@ -1,5 +1,17 @@
-all:
-	-@echo "Please use ./configure first.  Thank you."
+PSPSDK=$(shell psp-config --pspsdk-path)
+PSPDIR=$(shell psp-config --psp-prefix)
+TARGET_LIB = libz.a
+OBJS = adler32.o compress.o crc32.o deflate.o gzclose.o gzlib.o \
+       gzread.o gzwrite.o infback.o inffast.o inflate.o inftrees.o \
+       trees.o uncompr.o zutil.o
 
-distclean:
-	make -f Makefile.in distclean
+CFLAGS = -Wall -O2 -G0
+
+include $(PSPSDK)/lib/build.mak
+
+install: $(TARGET_LIB)
+	@echo "Installing libz into $(PSPDIR)"
+	@mkdir -p $(PSPDIR)/include $(PSPDIR)/lib
+	@cp zlib.h zconf.h $(PSPDIR)/include
+	@cp libz.a  $(PSPDIR)/lib
+	@echo "Done"
diff -Naur zlib-1.2.8/README.PSP zlib-1.2.8-psp/README.PSP
--- zlib-1.2.8/README.PSP	1969-12-31 19:00:00.000000000 -0500
+++ zlib-1.2.8-psp/README.PSP	2014-06-11 21:15:42.530582880 -0400
@@ -0,0 +1,4 @@
+To install, run:
+
+    make
+    make install
